roles/docker/tasks/main.yml

```diff
@@ -69,6 +69,22 @@
     reg_flag: --insecure-registry
   notify:
   - restart docker
+  
+- name: Set Proxy Settings
+  lineinfile:
+    dest: /etc/sysconfig/docker
+    regexp: '^{{ item.reg_conf_var }}=.*$'
+    line: "{{ item.reg_conf_var }}='{{ item.reg_fact_val }}'"
+  with_items:
+    - reg_conf_var: HTTP_PROXY
+      reg_fact_val: "{{ openshift.common.http_proxy | default('') }}"
+      when: "http_proxy in openshift.common"
+    - reg_conf_var: HTTPS_PROXY
+      reg_fact_val: "{{ openshift.common.https_proxy | default('') }}"
+    - reg_conf_var: NO_PROXY
+      reg_fact_val: "{{ openshift.common.no_proxy | default('') | join(',') }}"
```

Comment:
@sdodson we don't want to pull in openshift_common to avoid installing the base package for non-masters and non-nodes that are containerized (etcd is the current case, but as we containerize other roles they will factor in as well). We already set some common facts in openshift_facts, we should push additional ones we need here there as well.